src/lib.rs

```diff
@@ -325,7 +325,15 @@ impl OpenOptions {
     /// restricted.
     ///
     /// AIX, DragonFlyBSD, iOS, MacOSX, NetBSD, and OpenBSD: Not implemented as O_PATH is not defined.
-    #[cfg(not(any(target_os = "aix", target_os = "dragonfly", target_os = "ios", target_os = "macos", target_os = "netbsd", target_os = "openbsd")))]
+    #[cfg(not(any(
+        target_os = "aix",
+        target_os = "dragonfly",
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "illumos"
+    )))]
```

Comment:
Please update the documentation comment too.

Reply:
Done.